python/ModelTools.py

```diff
@@ -97,6 +97,23 @@ def doModel(self):
                 print "Wrote GraphVizTree of model_s to ",self.options.out+".dot"
 
     def doRateParams(self):
+
+    	# First support external functions/parameters 
+	for rp in self.DC.rateParams.keys():
+	 for rk in range(len(self.DC.rateParams[rp])):
+	  type = self.DC.rateParams[rp][rk][0][-1]
+	  if type!=2: continue
+	  argu,argv = self.DC.rateParams[rp][rk][0][0],self.DC.rateParams[rp][rk][0][1]
+	  if self.out.arg(argu): continue
+	  fin,wsn = argv.split(":")
+	  try:
+	    fitmp = ROOT.TFile.Open(fin)
+	    wstmp = fitmp.Get(wsn)
```

Comment:
I guess this could be slow if these input workspaces are large and/or there are many rateParam lines all needing something from the same workspace. We could probably keep the workspaces open for the duration of this function in this case, but could be implemented later